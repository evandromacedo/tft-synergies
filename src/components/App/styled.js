import { createGlobalStyle } from 'styled-components';
import BeaufortRegular from '../../static/font/Beaufort-Regular.ttf';
import BeaufortBold from '../../static/font/Beaufort-Bold.ttf';

export const GlobalStyle = createGlobalStyle`
  /* Resets */
  *, *:before, *:after {
    -webkit-box-sizing: border-box;
    -moz-box-sizing: border-box;
    box-sizing: border-box;
  }

  html, body, div, span, applet, object, iframe,
  h1, h2, h3, h4, h5, h6, p, blockquote, pre,
  a, abbr, acronym, address, big, cite, code,
  del, dfn, em, img, ins, kbd, q, s, samp,
  small, strike, strong, sub, sup, tt, var,
  b, u, i, center,
  dl, dt, dd, ol, ul, li,
  fieldset, form, label, legend,
  table, caption, tbody, tfoot, thead, tr, th, td,
  article, aside, canvas, details, embed,
  figure, figcaption, footer, header, hgroup,
  menu, nav, output, ruby, section, summary,
  time, mark, audio, video {
    margin: 0;
    padding: 0;
    border: 0;
    vertical-align: baseline;
  }

  body {
    -webkit-tap-highlight-color: transparent;
    -webkit-font-smoothing: antialiased;
    -moz-osx-font-smoothing: grayscale;
    line-height: 1;
  }

  ol, ul {
    list-style: none;
  }

  a {
    color: inherit;
    text-decoration: inherit;
  }

  input {
    outline-width: 0.3em;
    outline-color: transparent;
    -webkit-appearance: none;
    appearance: none;
    -webkit-border-radius:0;
    border-radius:0;
  }

  .input:disabled {
    background-color: transparent;
  }

  button {
    outline: none;
    cursor: pointer;
    background-color: transparent;
    margin: 0;
  }

  input,
  button {
    text-transform: inherit;
    font-family: inherit;
    letter-spacing: inherit;
    border: none;
  }

  /* Responsiveness */
  img,
  iframe,
  object,
  embed,
  video {
    height: auto;
    max-width: 100%;
  }

  /* Fonts */
  @font-face {
    font-family: 'Beaufort';
    src: url(${BeaufortRegular}) format('truetype');
    font-weight: normal;
    font-style: normal;
  }

  @font-face {
    font-family: 'Beaufort';
    src: url(${BeaufortBold}) format('truetype');
    font-weight: bold;
    font-style: normal;
  }

  html {
    font-family: 'Beaufort', sans-serif;
    font-weight: bold;
    font-size: 10px;
    color: #fff;
    background-color: #121212;
  }
`;
