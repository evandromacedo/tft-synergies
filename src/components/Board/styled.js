import styled from 'styled-components';

export const Wrapper = styled.section`
  display: grid;
  grid-gap: 2.4rem;
`;

export const BoardChampions = styled.div`
  display: grid;
  grid-gap: 2.4rem;
  grid-template-columns: repeat(auto-fill, minmax(20rem, 1fr));
  justify-content: space-evenly;

  @media (max-width: 856px) {
    grid-template-columns: repeat(auto-fill, minmax(15rem, 1fr));
  }

  @media (max-width: ${props => props.theme.extraSmall}) {
    justify-items: center;
  }
`;

export const Text = styled.p`
  ${props => props.theme.sans16pxItalic}
  text-align: center;
  color: rgba(255, 255, 255, 0.8);
`;
