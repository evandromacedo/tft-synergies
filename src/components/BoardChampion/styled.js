import styled from 'styled-components';
import TierBorder from '../Common/TierBorder';

export const Wrapper = styled(TierBorder)`
  position: relative;
  height: 12.8rem;
  padding: 0.8rem;
  max-width: 25.6rem;
  display: grid;
  grid-gap: 0.8rem;

  /* Background image for Splash Art */
  &:before {
    content: '';
    position: absolute;
    width: 100%;
    height: 100%;
    z-index: -1;
    background-image: url(${props => props.background});
    background-position: center right;
    background-size: cover;
  }
`;

export const Synergies = styled.div`
  display: grid;
  grid-gap: 0.8rem;
  grid-template-rows: repeat(3, 2rem);
`;

export const Items = styled.div`
  display: grid;
  grid-gap: 0.8rem;
  grid-template-columns: 2.8rem 2.8rem 2.8rem 1fr;
  align-items: center;
`;

export const Item = styled.div`
  width: 2.8rem;
  height: 2.8rem;
  background-color: ${props => props.theme.background3};
`;

export const Name = styled.p`
  ${props => props.theme.bold20px}
  text-align: center;
  text-overflow: ellipsis;
  white-space: nowrap;
  overflow: hidden;
`;
