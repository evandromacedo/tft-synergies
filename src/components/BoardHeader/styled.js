import styled from 'styled-components';

export const Wrapper = styled.div`
  display: grid;
  grid-template-columns: 1fr auto;
  align-items: center;

  @media (max-width: ${props => props.theme.extraSmall}) {
    grid-template-columns: 1fr;
    grid-gap: 1.2rem;
    justify-items: center;
  }
`;
