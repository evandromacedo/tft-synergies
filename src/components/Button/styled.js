import styled from 'styled-components';

export const Button = styled.button`
  ${props => props.theme.bold16px}
  text-transform: uppercase;
  color: inherit;
  padding: 0.7rem 2rem 0.5rem 2rem;
  border-radius: 2px;
  border-top: 2px solid ${props => props.theme.accentYellow};
  border-bottom: 2px solid #67502d;
  background-image: linear-gradient(
      to bottom,
      ${props => props.theme.accentYellow},
      #67502d
    ),
    linear-gradient(${props => props.theme.accentYellow}, #67502d);
  background-size: 2px 100%;
  background-position: 0 0, 100% 0;
  background-repeat: no-repeat;

  &:hover {
    filter: brightness(130%);
  }
`;

export const Content = styled.span`
  transform: translateY(1px);
`;
