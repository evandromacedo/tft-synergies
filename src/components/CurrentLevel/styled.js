import styled from 'styled-components';

export const Wrapper = styled.div`
  width: auto;
  display: grid;
  grid-gap: 1.2rem;
  justify-content: start;
  grid-auto-flow: column;
  align-items: center;
`;

export const Text = styled.p`
  ${props => props.theme.bold20px}
`;

export const Button = styled.button`
  padding: 0;

  @media (hover: hover) {
    &:hover {
      filter: brightness(130%);
    }
  }
`;
