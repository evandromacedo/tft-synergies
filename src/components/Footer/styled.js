import styled from 'styled-components';
import { Link as RouterLink } from 'react-router-dom';

export const Wrapper = styled.footer`
  max-width: ${props => props.theme.large};
  margin: 0 auto;
  padding: 2rem 2.4rem;
  display: grid;
  grid-gap: 2.4rem;
  grid-template-columns: auto auto;
  justify-content: space-between;
  align-items: center;
`;

export const Link = styled(RouterLink)`
  ${props => props.theme.bold20px}
  position: relative;

  @media (hover: hover) {
    &::before {
      content: ' ';
      position: absolute;
      bottom: -6px;
      height: 2px;
      width: 0;
      border-radius: 2px;
      background-color: ${props => props.theme.accentYellow};
      transition: width 0.2s;
    }

    &:hover::before {
      width: 100%;
    }
  }
`;

export const Contribute = styled(Link)`
  ${props => props.theme.bold20px}
  display: grid;
  grid-gap: 0.8rem;
  align-items: center;
  grid-template-columns: auto 2.4rem;

  span {
    transform: translateY(2px);
  }

  @media (max-width: ${props => props.theme.small}) {
    span {
      display: none;
    }

    grid-template-columns: 2.4rem;
  }
`;
