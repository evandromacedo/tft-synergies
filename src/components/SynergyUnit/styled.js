import styled from 'styled-components';

export const Wrapper = styled.div`
  height: 4rem;
  position: relative;
  /* Makes the block div to be width: auto */
  float: left;
  clear: left;
`;

export const Icon = styled.div`
  display: grid;
  position: relative;
  z-index: 1;

  /* Just to do the translate transition */
  background-image: linear-gradient(to right, #192128 2rem, transparent 0);
  background-repeat: no-repeat;
`;

export const Count = styled.div`
  cursor: default;
  display: grid;
  grid-auto-flow: column;
  grid-gap: 0.8rem;
  align-items: center;
  height: 4rem;
  padding: 0.6rem 0.8rem 0.6rem 2.4rem;
  background-color: rgba(255, 255, 255, 0.1);
  border-top-right-radius: 3px;
  border-bottom-right-radius: 3px;

  /* Transition to show or hide based on screen size */
  transform: translateX(2rem) translateY(-4rem);
  transition: transform 0.4s ease;

  &.hide {
    transform: translateX(-100%) translateY(-4rem);
  }
`;

export const Quantity = styled.span`
  ${props => props.theme.bold16px}
  border-radius: 5px;
  padding: 0.6rem 0.5rem 0.4rem 0.5rem;
  background-color: rgba(255, 255, 255, 0.2);
`;

export const NoSynergies = styled.p`
  ${props => props.theme.sans16pxItalic}
`;

export const Name = styled.p`
  ${props => props.theme.bold16px}
`;

export const Progress = styled.p`
  ${props => props.theme.bold12px}
  transform: translateY(1px);
`;

export const Unable = styled.span`
  color: rgba(255, 255, 255, 0.3);
`;
