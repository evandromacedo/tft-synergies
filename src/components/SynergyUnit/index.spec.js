import React from 'react';
import { shallow } from 'enzyme';
import toJSON from 'enzyme-to-json';
import { SynergyUnit } from '.';
import SynergyIcon from '../SynergyIcon';
import * as S from './styled';

const mockDragonDetails = {
  key: 'dragon',
  name: 'Dragon',
  description: null,
  accentChampionImage:
    'https://cdn.blitz.gg/blitz/centered/AurelionSol_Splash_Centered_0.jpg',
  bonuses: [
    {
      needed: 2,
      effect: 'Dragons are immune to 83% Magic damage'
    }
  ]
};

const mockDemonDetails = {
  key: 'demon',
  name: 'Demon',
  description:
    'Demons basic attacks have a 40% chance to burn 20 mana from their target and return mana to the attacker.',
  accentChampionImage: 'https://cdn.blitz.gg/blitz/centered/Aatrox_Splash_Centered_0.jpg',
  bonuses: [
    {
      needed: 2,
      effect: 'Return 15 mana back to attacker'
    },
    {
      needed: 4,
      effect: 'Return 30 mana back to attacker'
    },
    {
      needed: 6,
      effect: 'Return 45 mana back to attacker'
    }
  ]
};

describe('<SynergyUnit />', () => {
  it('renders properly', () => {
    shallow(<SynergyUnit />);
  });

  it('renders "no synergies yet" icon and text when there\'s no champion selected', () => {
    const wrapper = shallow(<SynergyUnit />);
    const NoSynergyIcon = wrapper.find(SynergyIcon);
    expect(NoSynergyIcon.exists()).toBe(true);

    const NoSynergyText = wrapper.find(S.NoSynergies);
    expect(NoSynergyText.text()).toMatch('No synergies yet');
  });

  it('renders the Quantity and Name properly', () => {
    const wrapper = shallow(<SynergyUnit count={5} details={mockDemonDetails} />);

    const QuantityText = wrapper.find(S.Quantity);
    expect(QuantityText.text()).toMatch('5');

    const NameText = wrapper.find(S.Name);
    expect(NameText.text()).toMatch(mockDemonDetails.name);
  });

  it("renders a PARTIAL Icon and inactive progress when there's no bonus yet", () => {
    const wrapper = shallow(
      <SynergyUnit count={1} details={mockDemonDetails} ranking="partial" />
    );
    const PartialIcon = wrapper.find(SynergyIcon);
    expect(PartialIcon.exists()).toBe(true);
    expect(PartialIcon.props().classOrOrigin).toEqual(mockDemonDetails.key);
    expect(PartialIcon.props().type).toEqual('partial');

    const ProgressText = wrapper.find(S.Progress);
    expect(ProgressText.text()).toMatch('1 / 2');
  });

  it('renders a BRONZE Icon and active progress when enables first bonus with exact quantity', () => {
    const wrapper = shallow(
      <SynergyUnit count={2} details={mockDemonDetails} ranking="bronze" />
    );
    const BronzeIcon = wrapper.find(SynergyIcon);
    expect(BronzeIcon.exists()).toBe(true);
    expect(BronzeIcon.props().classOrOrigin).toEqual(mockDemonDetails.key);
    expect(BronzeIcon.props().type).toEqual('bronze');

    const ProgressText = wrapper.find(S.Progress);
    expect(toJSON(ProgressText)).toMatchSnapshot();
  });

  it('renders a BRONZE Icon and active progress when enables first bonus with one more quantity', () => {
    const wrapper = shallow(
      <SynergyUnit count={3} details={mockDemonDetails} ranking="bronze" />
    );
    const BronzeIcon = wrapper.find(SynergyIcon);
    expect(BronzeIcon.exists()).toBe(true);
    expect(BronzeIcon.props().classOrOrigin).toEqual(mockDemonDetails.key);
    expect(BronzeIcon.props().type).toEqual('bronze');

    const ProgressText = wrapper.find(S.Progress);
    expect(toJSON(ProgressText)).toMatchSnapshot();
  });

  it('renders a SILVER Icon and active progress when enables middle bonus', () => {
    const wrapper = shallow(
      <SynergyUnit count={4} details={mockDemonDetails} ranking="silver" />
    );
    const SilverIcon = wrapper.find(SynergyIcon);
    expect(SilverIcon.exists()).toBe(true);
    expect(SilverIcon.props().classOrOrigin).toEqual(mockDemonDetails.key);
    expect(SilverIcon.props().type).toEqual('silver');

    const ProgressText = wrapper.find(S.Progress);
    expect(toJSON(ProgressText)).toMatchSnapshot();
  });

  it('renders a GOLD Icon and active number when enables solo bonus', () => {
    const wrapper = shallow(
      <SynergyUnit count={2} details={mockDragonDetails} ranking="gold" />
    );
    const GoldIcon = wrapper.find(SynergyIcon);
    expect(GoldIcon.exists()).toBe(true);
    expect(GoldIcon.props().classOrOrigin).toEqual(mockDragonDetails.key);
    expect(GoldIcon.props().type).toEqual('gold');

    const ProgressText = wrapper.find(S.Progress);
    expect(ProgressText.text()).toMatch('2');
  });

  it('renders a GOLD Icon and active progress when enables last bonus', () => {
    const wrapper = shallow(
      <SynergyUnit count={6} details={mockDemonDetails} ranking="gold" />
    );
    const GoldIcon = wrapper.find(SynergyIcon);
    expect(GoldIcon.exists()).toBe(true);
    expect(GoldIcon.props().classOrOrigin).toEqual(mockDemonDetails.key);
    expect(GoldIcon.props().type).toEqual('gold');

    const ProgressText = wrapper.find(S.Progress);
    expect(toJSON(ProgressText)).toMatchSnapshot();
  });
});
