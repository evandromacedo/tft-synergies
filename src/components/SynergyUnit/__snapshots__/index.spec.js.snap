// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<SynergyUnit /> renders a BRONZE Icon and active progress when enables first bonus with exact quantity 1`] = `
<styled.p>
  <span
    key="0"
  >
    2
    <styled.span>
       › 
    </styled.span>
  </span>
  <styled.span
    key="1"
  >
    4
     › 
  </styled.span>
  <styled.span
    key="2"
  >
    6
  </styled.span>
</styled.p>
`;

exports[`<SynergyUnit /> renders a BRONZE Icon and active progress when enables first bonus with one more quantity 1`] = `
<styled.p>
  <span
    key="0"
  >
    2
    <styled.span>
       › 
    </styled.span>
  </span>
  <styled.span
    key="1"
  >
    4
     › 
  </styled.span>
  <styled.span
    key="2"
  >
    6
  </styled.span>
</styled.p>
`;

exports[`<SynergyUnit /> renders a GOLD Icon and active progress when enables last bonus 1`] = `
<styled.p>
  <styled.span
    key="0"
  >
    2
     › 
  </styled.span>
  <styled.span
    key="1"
  >
    4
     › 
  </styled.span>
  <span
    key="2"
  >
    6
  </span>
</styled.p>
`;

exports[`<SynergyUnit /> renders a SILVER Icon and active progress when enables middle bonus 1`] = `
<styled.p>
  <styled.span
    key="0"
  >
    2
     › 
  </styled.span>
  <span
    key="1"
  >
    4
    <styled.span>
       › 
    </styled.span>
  </span>
  <styled.span
    key="2"
  >
    6
  </styled.span>
</styled.p>
`;
